java/test/src/main/java/test/Ice/idleTimeout/AllTests.java

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) ZeroC, Inc.
+
+package test.Ice.idleTimeout;
+
+import com.zeroc.Ice.Communicator;
+import com.zeroc.Ice.ConnectionClose;
+import com.zeroc.Ice.ConnectionLostException;
+import com.zeroc.Ice.ConnectionTimeoutException;
+import com.zeroc.Ice.InitializationData;
+import com.zeroc.Ice.Properties;
+import com.zeroc.Ice.Util;
+import java.io.PrintWriter;
+import test.Ice.idleTimeout.Test.TestIntfPrx;
+
+public class AllTests {
+  static void allTests(test.TestHelper helper) {
+    Communicator communicator = helper.communicator();
+    String proxyString = "test: " + helper.getTestEndpoint();
+    TestIntfPrx p = TestIntfPrx.uncheckedCast(communicator.stringToProxy(proxyString));
+
+    String proxyString3s = "test: " + helper.getTestEndpoint(1);
+
+    testIdleCheckDoesNotAbortConnectionWhenThreadPoolIsExhausted(p, helper.getWriter());
+    testConnectionAbortedByIdleCheck(proxyString, communicator.getProperties(), helper.getWriter());
+    testEnableDisableIdleCheck(
+        true, proxyString3s, communicator.getProperties(), helper.getWriter());
+    testEnableDisableIdleCheck(
+        false, proxyString3s, communicator.getProperties(), helper.getWriter());
+    p.shutdown();
+  }
+
+  // The client and server have the same idle timeout (1s) and the server enables connection idle
+  // checks (the default).
+  // We verify that the server's idle check does not abort the connection as long as this connection
+  // receives heartbeats, even when the heartbeats are not read off the connection in a timely
+  // manner.
+  // To verify this situation, we use an OA with a 1-thread thread pool and use this unique thread
+  // for a long synchronous dispatch (sleep).
+  private static void testIdleCheckDoesNotAbortConnectionWhenThreadPoolIsExhausted(
```

Comment:
To clarify how this works:

The only thread in the server thread pool is blocked in the dispatch of the `sleep` operation. This prevents the connection to read the heartbeats, and here the idle check doesn't abort the connection because the `isWaitingToBeRead` would return true.

Reply:
Yes, that's correct.